apps/condo/domains/user/schema/AuthenticateOrRegisterUserWithTokenService.test.js

```diff
@@ -0,0 +1,929 @@
+                const [confirmPhoneAction] = await createTestConfirmPhoneAction(adminClient, { isPhoneVerified: true })
+                const [result] = await authenticateOrRegisterUserWithTokenByTestClient(anonymousClient, {
+                    token: confirmPhoneAction.token,
+                    userType: SERVICE,
+                })
+                expect(result.token).not.toHaveLength(0)
+                expect(result.user.id).toBeDefined()
+                const user1 = await UserAdminUtils.getOne(adminClient, { id: result.user.id })
+
+                const anonymousClient2 = await makeClient()
+                const [confirmPhoneAction2] = await createTestConfirmPhoneAction(adminClient, {
+                    isPhoneVerified: true, phone: confirmPhoneAction.phone,
+                })
+                const [result2] = await authenticateOrRegisterUserWithTokenByTestClient(anonymousClient2, {
+                    token: confirmPhoneAction2.token,
+                    userType: SERVICE,
+                })
+                expect(result2.token).not.toHaveLength(0)
+                expect(result2.user.id).toBeDefined()
+                const user2 = await UserAdminUtils.getOne(adminClient, { id: result2.user.id })
+                expect(user1).toEqual(user2)
+            })
+
+            describe('if an staff or resident user with such a phone is already registered', () => {
+                let staffUser, resident, phone
+
+                beforeEach(async () => {
+                    phone = createTestPhone();
+                    [staffUser] = await createTestUser(adminClient, {
+                        type: STAFF,
+                        phone: phone,
+                    });
+                    [resident] = await createTestUser(adminClient, {
+                        type: RESIDENT,
+                        phone: phone,
+                    })
+                })
+
+                test('should register and sign in a service with a verified phone token', async () => {
+                    const [confirmPhoneAction] = await createTestConfirmPhoneAction(adminClient, {
+                        isPhoneVerified: true, phone: phone,
+                    })
+                    const [result] = await authenticateOrRegisterUserWithTokenByTestClient(anonymousClient, {
+                        token: confirmPhoneAction.token,
+                        userType: SERVICE,
+                    })
+                    expect(result.token).not.toHaveLength(0)
+                    expect(result.user.id).toBeDefined()
+                    const user = await UserAdminUtils.getOne(adminClient, { id: result.user.id })
+                    expect(user.id).not.toBe(staffUser.id)
+                    expect(user.type).toBe(SERVICE)
+                    expect(user.phone).toBe(phone)
+                })
+
+                test('should sign in a service with a verified phone token if a service with such a phone is already registered', async () => {
```

Comment:
How is this test different from the `should sign in service with confirmed phone token if a service with such a phone is already registered` on line 378? It seems to be the same.

Reply:
the difference is that in one case only one type of user is registered for a phone number, and in the other - all three